sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs

```diff
@@ -898,6 +1196,146 @@ internal virtual void SetNameFieldsIfNull()
         }
         #endregion Delete
 
+        #region Delete If Exists
+        /// <summary>
+        /// The <see cref="DeleteIfExists"/> operation marks the specified path
+        /// for deletion, if the path exists. The path is later deleted during
+        /// garbage collection.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/delete" />.
+        /// </summary>
+        /// <param name="recursive">
+        /// Required and valid only when the resource is a directory. If "true", all paths beneath the directory will be deleted.
+        /// If "false" and the directory is non-empty, an error occurs.
+        /// </param>
+        /// <param name="conditions">
+        /// Optional <see cref="DataLakeRequestConditions"/> to add conditions on
+        /// deleting this path.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> on successfully deleting.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response<bool> DeleteIfExists(
+            bool? recursive = default,
```

Comment:
Is it worth making this a regular bool with a default of `false` or `true`?  I don't have any good intuition what `null` would mean here as a customer.

Reply:
This is by design.   ```recursive``` isn't a valid parameter when deleting a File, so when it is set to default, it is not included.